drivers/storage/portworx/component/px_resource_gateway.go

```diff
@@ -0,0 +1,444 @@
+	var existingCommand []string
+	var existingEnvs []v1.EnvVar
+	for _, c := range existingDeployment.Spec.Template.Spec.Containers {
+		if c.Name == PxResourceGatewayContainerName {
+			existingImage = c.Image
+			existingCommand = c.Command
+			existingEnvs = append([]v1.EnvVar{}, c.Env...)
+			sort.Sort(k8sutil.EnvByName(existingEnvs))
+			break
+		}
+	}
+
+	targetDeployment := p.getPxResourceGatewayDeploymentSpec(cluster, ownerRef, imageName,
+		command, targetCPUQuantity, targetCPULimitQuantity)
+
+	// Check if the deployment has changed
+	modified := existingImage != imageName ||
+		!reflect.DeepEqual(existingCommand, command) ||
+		util.HasResourcesChanged(existingDeployment.Spec.Template.Spec.Containers[0].Resources, targetDeployment.Spec.Template.Spec.Containers[0].Resources) ||
+		util.HasNodeAffinityChanged(cluster, existingDeployment.Spec.Template.Spec.Affinity) ||
+		util.HaveTolerationsChanged(cluster, existingDeployment.Spec.Template.Spec.Tolerations)
+	if !p.isCreated || modified {
+		if err = k8sutil.CreateOrUpdateDeployment(p.k8sClient, targetDeployment, ownerRef); err != nil {
+			return err
+		}
+	}
+	p.isCreated = true
+	return nil
+}
+
+func (p *pxResourceGateway) createService(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	return k8sutil.CreateOrUpdateService(
+		p.k8sClient,
+		&v1.Service{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:            PxResourceGatewayServiceName,
+				Namespace:       cluster.Namespace,
+				OwnerReferences: []metav1.OwnerReference{*ownerRef},
+			},
+			Spec: v1.ServiceSpec{
+				Type: v1.ServiceTypeClusterIP,
+				Selector: map[string]string{
+					"name": PxResourceGatewayLabelName,
+				},
+				Ports: []v1.ServicePort{
+					{
+						Name:       PxResourceGatewayPortName,
+						Port:       PxResourceGatewayServicePort,
+						TargetPort: intstr.FromInt(PxResourceGatewayDeploymentPort),
+						Protocol:   v1.ProtocolTCP,
+					},
+				},
+			},
+		},
+		ownerRef,
+	)
+}
+
+func (p *pxResourceGateway) getPxResourceGatewayDeploymentSpec(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+	imageName string,
+	command []string,
+	cpuQuantity resource.Quantity,
+	cpuLimitQuantity resource.Quantity,
+) *appsv1.Deployment {
+	deploymentLabels := map[string]string{
+		"name": PxResourceGatewayLabelName,
+		"tier": "control-plane",
+	}
+	templateLabels := map[string]string{
+		"name": PxResourceGatewayLabelName,
+		"tier": "control-plane",
+	}
+	selectorLabels := map[string]string{
+		"name": PxResourceGatewayLabelName,
+		"tier": "control-plane",
+	}
+
+	replicas := int32(1)
+	maxUnavailable := intstr.FromInt(1)
+	maxSurge := intstr.FromInt(1)
+	imagePullPolicy := pxutil.ImagePullPolicy(cluster)
+
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      PxResourceGatewayDeploymentName,
+			Namespace: cluster.Namespace,
+			Annotations: map[string]string{
+				"scheduler.alpha.kubernetes.io/critical-pod": "",
```

Comment:
We don't need this. K8s doesn't look at this anymore. Instead they use priority class.